### ISSUE TITLE
Freezes during plan scan

### ISSUE THREAD

**Issue by Coalado** [seikur0/PGO-mapscan-opt#264]
Sometimes during plan scan, the scanner suddently freezes and does nothing. There is no error message or stacktrace. 

His last words:

[+] Login refresh.
[+](13) Weedle visible for 144 seconds (4481m NE from you)
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+](13) Weedle visible for 378 seconds (4624m NW from you)
[+] Login refresh.
[+](13) Weedle visible for 718 seconds (4362m NW from you)
[+](19) Rattata visible for 269 seconds (4364m NW from you)
[+](13) Weedle visible for 695 seconds (4395m NW from you)
[+](16) Pidgey visible for 589 seconds (4352m NW from you)
[+](16) Pidgey visible for 695 seconds (4376m NW from you)
[+](19) Rattata visible for 234 seconds (4449m NW from you)
[+](19) Rattata visible for 11 seconds (4396m NW from you)
[+](46) Paras visible for 585 seconds (4399m NW from you)
[+] Login refresh.
[+] Login refresh.
[+](32) Nidoran (m) visible for 648 seconds (4842m NW from you)
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+] Login refresh.
[+](19) Rattata visible for 901 seconds (4632m NW from you)
[+] Login refresh.
[+] Login refresh.
[94] auth/token error, refreshing login...
[+](43) Oddish visible for 901 seconds (6739m NW from you)
[96] auth/token error, refreshing login...

**Comment by seikur0** [comment]
Are there any kinds of bugs in the messages before somewhere?

**Comment by Coalado** [comment]
The Login isses #262 after startup
But it finished 2 plans until it freezed

Btw: it would be greta to add Time/Date to the std out.

**Comment by ghost** [comment]
I get this problem if I try to increase my scan range. I have not been able to finish a 2000 m scan. I can only get learn files from ~1000 m scans.

Edit:
I got it to work by letting the console alone for the whole hour (didn't even run the server). I kept scrolling and messing with the window on previous attempts.

**Comment by Zyrix** [comment]
I can confirm the freeze when using more accounts/greater scan range. After a couple of tries I was able to finish scanning for the spawn points and after that iscan seems to be more stable. However I still had a couple of freezes.

**Comment by seikur0** [comment]
Are you all on the newest version?

**Comment by Zyrix** [comment]
Yes, on the latest commit. I haven't seen any pattern as to when the scanner freezes. In order to bypass the problem I wrote a small script to restart the scanner once it hasn't written any new data for 5 minutes:

http://pastebin.com/4BM9HQXk

I added it to my crontab to run every 5 minutes:
*/5 \* \* \* \* ~/PGO-mapscan-opt/check_scanner.sh

**Comment by seikur0** [comment]
You should have an error somewhere, if that happens. @Coalado had the error #262, which crashed one of his accounts. It doesn't really depend on scan range or not touching the command window. In case it's another sqlite thing, I added some error messages for that.

**Comment by Coalado** [comment]
I still have this problem on my windows machine - without any error before.
However it runs fine on my ubuntu

**Comment by seikur0** [comment]
Try the more error messages update.
